src/main/java/appeng/container/implementations/ContainerFluidTerminal.java

```diff
@@ -0,0 +1,479 @@
+ * @since rv6 12/05/2018
+ */
+public class ContainerFluidTerminal extends AEBaseContainer implements IConfigManagerHost, IConfigurableObject, IMEMonitorHandlerReceiver<IAEFluidStack>
+{
+	private final IConfigManager clientCM;
+	private final IMEMonitor<IAEFluidStack> monitor;
+	private final IItemList<IAEFluidStack> fluids = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+	private PartFluidTerminal terminal;
+	private IConfigManager serverCM;
+	private IConfigManagerHost gui;
+	private IGridNode networkNode;
+
+	@GuiSync( 99 )
+	public boolean hasPower = false;
+
+	// Holds the fluid the client wishes to extract, or null for insert
+	private IAEFluidStack clientRequestedTargetFluid = null;
+
+	public ContainerFluidTerminal( InventoryPlayer ip, PartFluidTerminal terminal )
+	{
+		super( ip, terminal );
+		this.terminal = terminal;
+		this.clientCM = new ConfigManager( this );
+
+		this.clientCM.registerSetting( Settings.SORT_BY, SortOrder.NAME );
+		this.clientCM.registerSetting( Settings.SORT_DIRECTION, SortDir.ASCENDING );
+		this.clientCM.registerSetting( Settings.VIEW_MODE, ViewItems.ALL );
+		if( Platform.isServer() )
+		{
+			this.serverCM = terminal.getConfigManager();
+			this.monitor = terminal.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+			if( this.monitor != null )
+			{
+				this.monitor.addListener( this, null );
+				final IGridNode node = terminal.getGridNode( AEPartLocation.INTERNAL );
+				if( node != null )
+				{
+					this.networkNode = node;
+					final IGrid g = node.getGrid();
+					if( g != null )
+					{
+						this.setPowerSource( new ChannelPowerSrc( this.networkNode, g.getCache( IEnergyGrid.class ) ) );
+					}
+				}
+			}
+		}
+		else
+		{
+			this.monitor = null;
+		}
+		this.bindPlayerInventory( ip, 0, 222 - 82 );
+	}
+
+	@Override
+	public boolean isValid( Object verificationToken )
+	{
+		return true;
+	}
+
+	@Override
+	public void postChange( IBaseMonitor<IAEFluidStack> monitor, Iterable<IAEFluidStack> change, IActionSource actionSource )
+	{
+		for( final IAEFluidStack is : change )
+		{
+			this.fluids.add( is );
+		}
+	}
+
+	@Override
+	public void addListener( IContainerListener listener )
+	{
+		super.addListener( listener );
+
+		this.queueInventory( listener );
+	}
+
+	@Override
+	public void onListUpdate()
+	{
+		for( final IContainerListener c : this.listeners )
+		{
+			this.queueInventory( c );
+		}
+	}
+
+	private void queueInventory( final IContainerListener c )
+	{
+		if( Platform.isServer() && c instanceof EntityPlayer && this.monitor != null )
+		{
+			try
+			{
+				PacketMEFluidInventoryUpdate piu = new PacketMEFluidInventoryUpdate();
+				final IItemList<IAEFluidStack> monitorCache = this.monitor.getStorageList();
+
+				for( final IAEFluidStack send : monitorCache )
+				{
+					try
+					{
+						piu.appendFluid( send );
+					}
+					catch( final BufferOverflowException boe )

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)